Minor flow improvement for binary optimization how-to (#582)
OffchainLabs/arbitrum-docs: arbitrum-docs/stylus/how-tos/optimizing-binaries.mdx

<PublicPreviewBannerPartial />

> To be deployed onchain, the size of your **uncompressed WebAssembly (WASM) file** must not exceed 128Kb, while the **compressed binary** must not exceed 24KB.

Reviewer: ```suggestion
To be deployed onchain, the size of your **uncompressed WebAssembly (WASM) file** must not exceed 128Kb, while the **compressed binary** must not exceed 24KB. Stylus conforms with the same contract size limit as the EVM to remain fully interoperable with all smart contracts on Arbitrum chains.
```

cargo stylus deploy --nightly
```

## Third-party optimization tooling

Additional WASM-specific tooling exists to shrink binaries. Due to being 3rd party, users should use these at their own risk.

### `wasm-opt`

[wasm-opt](https://docs.rs/wasm-opt/0.113.0/wasm_opt/) applies techniques to further reduce binary size, usually netting around 10%.

### Frequently asked questions

**Will future releases of Stylus introduce additional optimizations?**

Yes! We're actively working on improving WASM sizes generated by Rust code with the Stylus SDK.

**Why don't I have to worry about this type of optimization when I use `cargo` without using Stylus?**

On modern platforms, tools like `cargo` don’t have to worry about the size of the binaries they produce. This is because there’s many orders of magnitude more storage available than even the largest of binaries, and for most applications it’s media like images and videos that constitutes the majority of the footprint.

> Nevertheless, systems programming languages compete for viability in the OS and embedded space, where resource constraints are extremely strict. Hence, while not the default options, tooling often provides mechanisms for reducing binary bloat, such as the options outlined in this document.

Reviewer: ```suggestion
Resource constraints when building on blockchains are extremely strict. Hence, while not the default option, tooling often provides mechanisms for reducing binary bloat, such as the options outlined in this document.
```
